client/landing/gutenboarding/components/domain-picker/list.tsx

```diff
@@ -17,8 +17,11 @@ import { useDebounce } from 'use-debounce';
 /**
  * Internal dependencies
  */
-import { DomainSuggestion, DomainSuggestionQuery } from '../../stores/domain-suggestions/types';
-import { STORE_KEY } from '../../stores/domain-suggestions';
+import {
+	DomainSuggestion,
+	DomainSuggestionQuery,
+} from '@automattic/calypso-data-stores/src/domain-suggestions/types';
```

Comment:
Might expose these in a nicer way? Would probably mean having

```ts
export * from './domains-suggestions/types';
export * from './verticals/types';
export * from './verticals-templates/types';
```

in the package's `index.ts`

Reply:
> I don't think the example will be possible because exported names will collide. Each of the stores exports `STORE_KEY`.

I was only going to export each store's `types` file here, so there shouldn't be any collisions. I _think_ that exporting those (plus the `STORE_KEY`s, as I'm already doing) should be enough, since that's basically our API: Users pretty much only need the store keys to access the stores, and the type defs to type the data obtained from the stores. Everything else -- reducers, actions, selectors, resolvers -- can be arguably deemed implementation and thus not publicly available.

> I do think this direction is preferable, I'd like to avoid having the API be to import `…/src/foo/bar` from packages as much as possible.

:+1: 

> I don't think namespace exports (`export * as myStore from './my-store'`) is available, which means we'd have to do `import * as myStore from './my-store'; export { myStore };`. There's a strong change that will complicate treeShaking.

See above -- only planning to export aliased store keys, plus types.

Your note about side effects is somewhat related though, I'll reply there.